assets/js/components/DashboardMainApp.js

```diff
@@ -85,6 +85,12 @@ export default function DashboardMainApp() {
 	const audienceSegmentationEnabled = useFeature( 'audienceSegmentation' );
 
 	const [ showSurveyPortal, setShowSurveyPortal ] = useState( false );
+	const currentSurvey = useSelect( ( select ) =>
+		select( CORE_SITE ).isUsingProxy() &&
+		select( CORE_USER ).areSurveysOnCooldown() === false
+			? select( CORE_USER ).getCurrentSurvey()
+			: null
+	);
```

Comment:
We should short-circuit this `useSelect()` callback to avoid making a potentially unneeded request to the `survey` endpoint on page load.

```suggestion
	const [ showSurvey, setShowSurvey ] = useState( false );
	const showCurrentSurvey = useSelect( ( select ) =>
		showSurvey &&
		select( CORE_SITE ).isUsingProxy() &&
		select( CORE_USER ).areSurveysOnCooldown() === false
			? select( CORE_USER ).getCurrentSurvey()
			: null
	);
```